task.sql

```diff
@@ -4,30 +4,56 @@ CREATE DATABASE ShopDB;
 USE ShopDB;
 
 CREATE TABLE Countries (
-    ID INT,
+    ID INT AUTO_INCREMENT,
     Name VARCHAR(50),
     PRIMARY KEY (ID)
 );
 
+CREATE TABLE Products (
+    ID INT AUTO_INCREMENT,
+    Name VARCHAR(50),
+    PRIMARY KEY (ID)
+);
+
+CREATE TABLE Warehouses (
+    ID INT AUTO_INCREMENT,
+    Name VARCHAR(50),
+    Address VARCHAR(50),
+    CountryID INT,
+    PRIMARY KEY (ID),
+    FOREIGN KEY (CountryID) REFERENCES Countries(ID) ON DELETE NO ACTION
+);
+
 CREATE TABLE ProductInventory (
-    ID INT,
-    ProductName VARCHAR(50),
+    ID INT AUTO_INCREMENT,
+    ProductID INT,
+    WarehouseID INT,
     WarehouseAmount INT,
-    WarehouseName VARCHAR(50),
-    WarehouseAddress VARCHAR(50), 
-    CountryID INT,
-	FOREIGN KEY (CountryID) REFERENCES Countries(ID) ON DELETE NO ACTION,
-    PRIMARY KEY (ID)
+    PRIMARY KEY (ID),
+    FOREIGN KEY (ProductID) REFERENCES Products (ID) ON DELETE NO ACTION,
+    FOREIGN KEY (WarehouseID) REFERENCES Warehouses (ID) ON DELETE NO ACTION 
```

Comment:
Similar to the previous comment, verify that `ON DELETE NO ACTION` is the intended behavior for these foreign key constraints. If you want to allow deletion of a product or warehouse and automatically handle the related entries in `ProductInventory`, consider using `ON DELETE CASCADE`.